### ISSUE TITLE
Enable fields in Google Sheets to set the variant of a component

### ISSUE THREAD

**Issue by jgpippin** [DWilliames/google-sheets-sync-figma#4]
When a design has components with variants, enable the Google Sheet to define the variant to use of the component.

I'd propose using either the slash naming convention described here: https://help.figma.com/hc/en-us/articles/360055471353-Prepare-for-Variants

or the array naming convention:
`Property 1 Value=Value, Property 2 Value=Value, Property 3 Value=Value`

This would be useful for use cases wherein certain rows in the sheet need to adjust the layout, or signal which data to include, without having to use /Hide and /Show to accomplish this.

**Comment by DWilliames** [comment]
Great idea. Thanks @jgpippin.
I think I have this working now for the next update.
The way it will work is using: `Property 1 Value=Value, Property 2 Value=Value, Property 3 Value=Value` as that's what the component is named under the hood in Figma.

**Comment by DWilliames** [comment]
Hi @jgpippin – I pushed an update to the plugin with a change to support this.
Are you able to check if it works for you now?

**Comment by jgpippin** [comment]
Fantastico! Excited to check it out. 

I don't see updated documentation; should I assume that as long as the component has the right #Title naming convention the data type should be /`Property 1 Value=Value, Property 2 Value=Value` with all values accounted for?

**Comment by DWilliames** [comment]
Yeah I haven't updated the documentation yet.
I'm planning to do a complete overhaul of the the docs.

But essentially that's correct. If you have a Component Instance in Figma with the name `#row` for example.
And then in your Google Sheet, make sure a value in the row/column named 'Row' has the exact name of the component variant you want it to update to, such as `Property 1 Value=Value, Property 2 Value=Value`.

Let me know how it goes.
I'll close of this issue for now — but if you encounter any bugs I'll re-open it.

**Comment by jgpippin** [comment]
Okay, I patched my Figma and Google Sheets files to use this new setup, but when I attempt to run the plugin I get the following error: 

`Error: in set_y: Expected "y" to have type number but got NaN instead`

I'm unclear whether the value in row should have a leading slash, so I tried each and got the same error each time.

**Comment by jgpippin** [comment]
Okay, I tried to adjust my variant naming so that no two components had the same naming convention:

- "Card Body" naming convention "Layout=Event"
- "Card Title" naming convention "Title=Event"

But I got the same error.

Just gonna drop a screenshot in here so we're clear on how I'm naming these in the file. I've checked that every value corresponds to the right variant property/name values.

<img width="212" alt="Screen Shot 2021-01-31 at 6 47 00 PM" src="https://user-images.githubusercontent.com/929287/106401896-be407f00-63f4-11eb-9a9f-71ad95768500.png">

**Comment by DWilliames** [comment]
Ah interesting, I thought I'd patched the `Error: in set_y` bug. But I think I may know what I need to do there – however, in theory that should be a separate issue.

Can you post a screenshot of your **Mains Component (Variant)** properties from the inspector like this? 
![image](https://user-images.githubusercontent.com/6570446/106402899-fd67e200-647f-11eb-8902-07709d1fb6d1.png)

At the moment you can't specify the value for just one variant Property at a time. You need to spell out all of them at once (as that's the name of the component layer).

So assuming it was something like this screenshot; for example the value `Type=Primary, Size=Large` wouldn't work. You'd need to spell out the entire thing: `Type=Primary, Size=Large, State=Default, Icon=False` in your Google Sheets cell.

**Comment by jgpippin** [comment]
> At the moment you can't specify the value for just one variant Property at a time. You need to spell out all of them at once (as that's the name of the component layer).

Yes, that's what I'm doing; I'm just using components that only had one property:

<img width="238" alt="Screen Shot 2021-01-31 at 7 30 25 PM" src="https://user-images.githubusercontent.com/929287/106403111-e92dd180-63fa-11eb-8c13-fc9ca860c5af.png">
<img width="230" alt="Screen Shot 2021-01-31 at 7 30 13 PM" src="https://user-images.githubusercontent.com/929287/106403118-edf28580-63fa-11eb-8a21-ab2a02724619.png">

**Comment by jgpippin** [comment]
Just wanted to check in here; this thread went dry and I'm not sure if you had what you needed to debug.
